server/routes/companies.js

```diff
@@ -116,25 +179,459 @@ router.get('/filter', function(req, res, next) {
+  let user_ids = req.body.user_id;
+  let position_ids = req.body.position_id;
+  let major_ids = req.body.major_id;
+  let contact_ids = req.body.contact_id;
+  let event_ids = req.body.event_id;
+
+  let rankValues = [];
+  if (ranks && user_ids) {
+    if(Array.isArray(ranks) && Array.isArray(user_ids)){
+      let length = ranks.length;
+      if (ranks.length != user_ids.length)
+        util.throwError(400, "Both user_ids and ranks must be the same size.");
+      length = user_ids.length; 
+      for (let n=0; n < length; n++) {
+        rankValues.push({
+          user_id: user_ids[n],
+          rank: ranks[n]
+
+        });
+       // util.throwError(400, ranks)
+      }
+    } else if(Array.isArray(ranks) || Array.isArray(user_ids)) {
+       util.throwError(400, "Both user_ids and ranks must be the same size.");
+    }else{
+      rankValues.push({
+          user_id: user_ids,
+          rank: ranks
+        });
+    }
+
   }
+
+  let companyPositions = [];
+  if (position_ids) {
+    if(Array.isArray(position_ids)){
+      let length = position_ids.length;
+      for (let n=0; n < length; n++) {
+        companyPositions.push({
+           position_id: position_ids[n]
+        });
+      }
+    } else{
+      companyPositions.push({
+           position_id: position_ids
+      })
+    }
+  }
+
+  let companyMajors = [];
+  if (major_ids) {
+    if(Array.isArray(major_ids)){
+      let length = major_ids.length;
+      for (let n=0; n < length; n++) {
+        companyMajors.push({
+           major_id: major_ids[n]
+        })
+      }
+    } else {
+      companyMajors.push({
+           major_id: major_ids
+        })
+    }
+  }
+
+  let companyContacts = [];
+  if (contact_ids) {
+    if(Array.isArray(contact_ids)){
+      let length = contact_ids.length;
+      for (let n=0; n < length; n++) {
+        companyContacts.push({
+           contact_id: contact_ids[n]
+        })
+      }
+    } else {
+        companyContacts.push({
+           contact_id: contact_ids
+        })
+    }
+  }   
+
+  let companyEvents = [];
+  if (event_ids) {
+    if(Array.isArray(event_ids)){
+      let length = event_ids.length;
+      for (let n=0; n < length; n++) {
+        companyEvents.push({
+           event_id: event_ids[n]
+        })
+      }
+    } else {
+        companyEvents.push({
+           event_id: event_ids
+        })
+    }
+  }
+
+
+  var queryCompany = knex('company').insert(values)
+  knex.transaction(function(trx) {
+    return queryCompany.transacting(trx)
+      .returning('id')
+      .then(async function(ids){
+        let company_id = ids[0];
+
+        if (ranks && user_ids) {
+          rankValues.forEach(function(element){
+            element.company_id = company_id;
+          })
+          var queryRank = knex('user_company_rank').insert(rankValues)
+          await queryRank.transacting(trx);
+        }
+
+        if (position_ids) {
+          companyPositions.forEach(function(element){
```

Comment:
Make these forEach() lines one line for conciseness

e.g. companyPositions.forEach(function(element){  element.company_id = company_id; })